lib/common_test/src/ct_snmp.erl

```diff
@@ -122,33 +122,51 @@
-%%% @type call_back_module() = atom()
-%%% @type user_data() = term() 
-%%% @type oids() = [oid()]
-%%% @type oid() = [byte()]
-%%% @type snmpreply() = {error_status(), error_index(), varbinds()} 
-%%% @type error_status() = noError | atom() 
-%%% @type error_index() = integer() 
-%%% @type varbinds() = [varbind()] 
-%%% @type varbind() =  term() 
-%%% @type value_type() = o ('OBJECT IDENTIFIER') | i ('INTEGER') | 
-%%% u ('Unsigned32') | g ('Unsigned32') | s ('OCTET STRING') 
-%%% @type varsandvals() = [var_and_val()]
-%%% @type var_and_val() = {oid(), value_type(), value()}
-%%% @type sec_type() = none | minimum | semi
-%%% @type rel_path() = string() 
-%%% @type snmp_app_manager_params() = term()
-%%% @type snmp_app_agent_params() = term()
+-type ip() :: string() | {integer(), integer(), integer(), integer()}.
+-type agent_ip() :: ip().
+-type manager_ip() :: ip().
+-type agent_name() :: atom().
+-type agent_port() :: integer().
+-type agent_config() :: {start_agent, boolean()} |
+                        {agent_sysname, string()} |
+                        {agent_manager_ip, manager_ip()} |
+                        {agent_vsns, list()} |
+                        {agent_trap_udp, integer()} |
+                        {agent_udp, integer()} |
+                        {agent_notify_type, atom()} |
+                        {agent_sec_type, sec_type()} |
+                        {agent_passwd, string()} |
+                        {agent_engine_id, string()} |
+                        {agent_max_msg_size, string()} |
+                        {agent_contexts, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_community, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_sysinfo,  [term()] | {data_dir_file, rel_path()}} |
+                        {agent_vacm, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_usm, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_notify_def, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_target_address_def, [term()] | {data_dir_file, rel_path()}} |
+                        {agent_target_param_def, [term()] | {data_dir_file, rel_path()}}.
+-type user_name() :: atom().
+-type agent() :: {agent_name(), [user_name() | agent_ip() | agent_port() | [agent_config()]]}.
+-type user_data() :: term().
+-type call_back_module() :: atom().
+-type user() :: {user_name(), [call_back_module() | user_data()]}.
+-type usm_user_name() :: string().
+-type usm_config() :: [term()] | {data_dir_file, rel_path()}.
+-type usm_user() :: {usm_user_name(), [usm_config()]}.
+-type oids() :: [oid()].
+-type oid() :: [byte()].
+-type snmpreply() :: {error_status(), error_index(), varbinds()}.
+-type error_status() :: noError | atom().
+-type error_index() :: integer().
+-type varbinds() :: [varbind()].
+-type varbind() :: term().
+-type value_type() :: o | i | u | g | s | b | ip | op | c32 | c64 | tt.
```

Comment:
You removed the information about what each letter means (o('OBJECT IDENTIFIER') | i('INTEGER') | u('Unsigned32') | g('Unsigned32') | s('OCTET STRING')). Maybe this could be added in a comment?